etl/steps/data/garden/ember/2022-12-13/yearly_electricity.py

```diff
@@ -0,0 +1,387 @@
+REGIONS = {
+    # Default continents.
+    "Africa": {},
+    "Asia": {},
+    "Europe": {},
+    "European Union (27)": {},
+    "North America": {},
+    "Oceania": {},
+    "South America": {},
+    # Ember already has data for "World".
+    # "World": {},
+    # Income groups.
+    "Low-income countries": {},
+    "Upper-middle-income countries": {},
+    "Lower-middle-income countries": {},
+    "High-income countries": {},
+}
+
+# Conversion factors.
+# Terawatt-hours to kilowatt-hours.
+TWH_TO_KWH = 1e9
+# Megatonnes to grams.
+MT_TO_G = 1e12
+
+# Map units (short version) to unit name (long version).
+SHORT_UNIT_TO_UNIT = {
+    "TWh": "terawatt-hours",
+    "MWh": "megawatt-hours",
+    "kWh": "kilowatt-hours",
+    "mtCO2": "megatonnes of CO2 equivalent",
+    "gCO2/kWh": "grams of CO2 equivalent per kilowatt-hour",
+    "GW": "gigawatts",
+    "%": "%",
+}
+
+# Categories expected to exist in the data.
+CATEGORIES = [
+    "Capacity",
+    "Electricity demand",
+    "Electricity generation",
+    "Electricity imports",
+    "Power sector emissions",
+]
+
+# Choose columns for which region aggregates should be created.
+SUM_AGGREGATES = [
+    # "Bioenergy (%)",
+    "Bioenergy (GW)",
+    "Bioenergy (TWh)",
+    "Bioenergy (mtCO2)",
+    # "CO2 intensity (gCO2/kWh)",
+    # "Clean (%)",
+    "Clean (GW)",
+    "Clean (TWh)",
+    "Clean (mtCO2)",
+    # "Coal (%)",
+    "Coal (GW)",
+    "Coal (TWh)",
+    "Coal (mtCO2)",
+    "Demand (TWh)",
+    # "Demand per capita (MWh)",
+    # "Fossil (%)",
+    "Fossil (GW)",
+    "Fossil (TWh)",
+    "Fossil (mtCO2)",
+    # "Gas (%)",
+    "Gas (GW)",
+    "Gas (TWh)",
+    "Gas (mtCO2)",
+    "Gas and Other Fossil (%)",
+    "Gas and Other Fossil (GW)",
+    "Gas and Other Fossil (TWh)",
+    "Gas and Other Fossil (mtCO2)",
+    # "Hydro (%)",
+    "Hydro (GW)",
+    "Hydro (TWh)",
+    "Hydro (mtCO2)",
+    "Hydro, Bioenergy and Other Renewables (%)",
+    "Hydro, Bioenergy and Other Renewables (GW)",
+    "Hydro, Bioenergy and Other Renewables (TWh)",
+    "Hydro, Bioenergy and Other Renewables (mtCO2)",
+    "Net Imports (TWh)",
+    # "Nuclear (%)",
+    "Nuclear (GW)",
+    "Nuclear (TWh)",
+    "Nuclear (mtCO2)",
+    # "Other Fossil (%)",
+    "Other Fossil (GW)",
+    "Other Fossil (TWh)",
+    "Other Fossil (mtCO2)",
+    # "Other Renewables (%)",
+    "Other Renewables (GW)",
+    "Other Renewables (TWh)",
+    "Other Renewables (mtCO2)",
+    # "Renewables (%)",
+    "Renewables (GW)",
+    "Renewables (TWh)",
+    "Renewables (mtCO2)",
+    # "Solar (%)",
+    "Solar (GW)",
+    "Solar (TWh)",
+    "Solar (mtCO2)",
+    "Total Generation (TWh)",
+    "Total emissions (mtCO2)",
+    # "Wind (%)",
+    "Wind (GW)",
+    "Wind (TWh)",
+    "Wind (mtCO2)",
+    # "Wind and Solar (%)",
+    "Wind and Solar (GW)",
+    "Wind and Solar (TWh)",
+    "Wind and Solar (mtCO2)",
+]
+
+
+def load_global_electricity_review(tb_meadow: catalog.Table) -> pd.DataFrame:
```

Comment:
I'd probably use another name for the function, as the data is already loaded. Here it seems that you do some pre-processing and basic cleaning of the table, plus changing the type to pandas.DataFrame.

Also, the part with "global_electricity_review" can be confusing, as now you are loading "yearly_electricity" dataset.

Perhaps something like `load_yearly_electricity_as_df`.

Reply:
I renamed it `prepare_yearly_electricity_data`.